src/org/pahappa/systems/services/IncidentServiceImpl.java

```diff
@@ -1,14 +1,34 @@
 package org.pahappa.systems.services;
+import java.util.ArrayList;
+import java.util.Date;
 import java.util.List;
 
+import org.pahappa.systems.enums.Status;
+import org.pahappa.systems.exceptions.SavingFailedException;
 import org.pahappa.systems.models.Incident;
 
 public class IncidentServiceImpl implements IncidentService {
 
+	private  static List<Incident> incidents = new ArrayList<Incident>();
+	private static int incidentIds=0;
+
 	@Override
 	public Incident saveIncident(Incident incident) throws Exception {
-		// TODO Auto-generated method stub
-		return null;
+
+
+		if ((incident.getComment()!=null) & (incident.getTitle() != null)){
+			incident.setId(++incidentIds);
```

Comment:
For null values, the service explains that we shall throw a ValidationFailedException. I request you first finalize the exceptions pull request.
<img width="1056" alt="Screen Shot 2021-09-10 at 3 14 04 PM" src="https://user-images.githubusercontent.com/22131999/132855362-540b1818-f132-4b04-ba77-7cd177d95fd8.png">

Reply:
Finished

Reply:
@MutebiAlexKato, I don't think even this code does what it intends.
https://www.tutorialspoint.com/Differences-between-and-and-and-and-operators-in-Java

Also, cater for empty comments and strings